vividus-plugin-web-app/src/test/java/org/vividus/proxy/steps/ProxyStepsTests.java

```diff
@@ -261,6 +267,17 @@ void testWaitRequestInProxyLog()
                         httpMethod, URL_PATTERN).equals(e.toString())));
     }
 
+    @Test
+    void testAddHeadersToProxyRequest()
+    {
+        HttpRequest request = mock(HttpRequest.class);
+        HttpMessageContents contents = mock(HttpMessageContents.class);
+        HttpMessageInfo message = mock(HttpMessageInfo.class);
+        ExamplesTable headers = new ExamplesTable("|name|value|\n|name1|value1|");
+        proxySteps.addHeadersToProxyRequest(StringComparisonRule.IS_EQUAL_TO, URL_PATTERN, headers);
+        verify(proxy).addRequestFilter(argThat(filter -> filter.filterRequest(request, contents, message) == null));
```

Comment:
I believe unit test should be expanded a bit to invoke the filter actually (and 2 unit tests will be required: when URL matches and doesn't match to the rule)